apps/dash-phylogeny/utils.py

```diff
@@ -0,0 +1,743 @@
+                    thickness=20,
+                    len=0.7,
+                    autotick=False,
+                    titleside="right",
+                    title="Number of " + virus_name.title() + " Virus",
+                ),
+            )
+        ]
+
+        layout = dict(
+            title=virus_name.title()
+            + " Virus cases in West Africa <br>Between "
+            + str(min_date)
+            + " and "
+            + str(max_date),
+            margin=dict(l=0, r=0, b=0, t=35),
+            geo=dict(
+                showframe=False, showcoastlines=False, projection=dict(type="Mercator")
+            ),
+            autosize=True,
+            legend={"orientation": "h"},
+            font=dict(family="Open Sans"),
+        )
+
+        fig = dict(data=data, layout=layout)
+        return fig
+
+
+country_colors = [
+    "red",
+    "green",
+    "blue",
+    "orange",
+    "purple",
+    "pink",
+    "yellow",
+    "black",
+    "gray",
+    "silver",
+    "violet",
+    "yellowgreen",
+    "turquoise",
+    "sienna",
+    "salmon",
+]
+
+
+def create_curve_line(df, virus_name, min_date, max_date):
+    results_country_by_year = df.groupby(["Country", "Year"])["Value"].sum()
+    # Translate pandas.core.series.series object in dataframe
+    df_group_by_country = results_country_by_year.to_frame()
+    # Rename the first column in Value
+    df_group_by_country.columns = ["Value"]
+    # Move the index values (i.e. Country column) in column and reset index
+    df_group_by_country = df_group_by_country.reset_index()
+    country_list = df_group_by_country.Country.unique()
+    p_data = []
+    p_name = []
+    p_year = []
+
+    for country_name in country_list:
+        country_data = np.extract(
+            df_group_by_country.Country == country_name, df_group_by_country.Value
+        )
+        country_year = np.extract(
+            df_group_by_country.Country == country_name, df_group_by_country.Year
+        )
+
+        if country_year[0] < min_date:
+            min_date = country_year[0]
+        if country_year[len(country_year) - 1] > max_date:
+            max_date = country_year[len(country_year) - 1]
+
+        # Stock country name in array structure
+        if len(country_name) != 0:
+            p_name.append(country_name)
+
+        # Stock country data in array structure
+        if len(country_data) != 0:
+            p_data.append(country_data)
+
+        # Stock country year in array structure
+        if len(country_year) != 0:
+            p_year.append(country_year)
+
+    # Add step in x axe
+    step = 1
+    if 5 < max_date - min_date <= 10:
+        step = 2
+    elif 10 < max_date - min_date <= 50:
+        step = 5
+    elif max_date - min_date > 50:
+        step = 10
+    marks_data = []
+    for i in range(int(min_date), int(max_date + 1), step):
+        marks_data.append(str(i))
+
+    if i < int(max_date):
+        marks_data.append(str(max_date))
+
+    year = marks_data
```

Comment:
It doesn't look like this variable is needed for the rest of the function

Reply:
> ![Screen Shot 2020-01-09 at 2 35 39 PM](https://user-images.githubusercontent.com/30911511/72099927-a67cf100-32ef-11ea-9676-116ef67f3e86.png)
> 
> Is it okay to have 2014.5 on the graph there?

Fixed by tickformat to integer 💯